facturx/facturx.py

```diff
@@ -147,3 +147,129 @@ def write_xml(self, path):
         with open(path, 'wb') as f:
             f.write(self.xml_str)
 
+    def __make_dict(self):
+        field_file = os.path.join(os.path.dirname(__file__), 'flavors\\fields.yml')
+        req_list = []
+        with open(field_file, 'r') as f:
+            data = f.read()
+            dict_data = yaml.load(data)
+
+        for k, v in dict_data.items():
+            for kv, vv in v.items():
+                if kv == '_required':
+                    if vv is True:
+                        req_list.append(k)
+
+        factx_value = []
+        factx_key = []
+        zugf_key = []
+        zugf_value = []
+
+        for k, v in dict_data.items():
+            for item in req_list:
+                if k == item:
+                    for vk, vv in v.items():
+                        if vk == '_path':
+                            for vvk, vvv in vv.items():
+                                if vvk == 'factur-x':
+                                    factx_key.append(k)
+                                    factx_value.append(vvv)
+                                elif vvk == 'zugferd':
+                                    zugf_key.append(k)
+                                    zugf_value.append(vvv)
+
+        self.factx = {}
```

Comment:
It's not clear what the variable `factx` should mean. You have to be very precise or it will get confusing in the future.